test/controllers/sessions_controller_test.rb

```diff
@@ -0,0 +1,32 @@
+require "test_helper"
+
+describe SessionsController do
+  let(:user1) { users(:user1) }
+
+  it "should got to login page" do
+    get login_path
+    must_respond_with :success
+  end
+
+  describe "if a user exists" do
+    it "should find the user, pass the user to session and redirect" do
+      post login_path, params: { name: user1.username }
+      session[:logged_in_as_user].must_equal user1.id
+      must_respond_with :redirect
+    end
+  end
+
+  describe "if the user doesn't exist" do
+    it "should have an error in session" do
```

Comment:
This really isn't what the `describe` block describes.  Better would be, `describe 'if the user field is nil, or empty string the user is presented with an error message'`